oneclick.py

```diff
@@ -165,7 +165,15 @@ def request(self, url, method='get', extra_headers={}, data=None,
             allowed_response_codes=allowed_response_codes, 
             disallowed_response_codes=disallowed_response_codes
         )
-        
+    
+        if (response.content
+            and 'Invalid Basic Token or permission denied' in response.content):
+            raise BadResponseException(
+                url, "Permission denied. This may be a temporary rate-limiting issue, or the credentials for this collection may be wrong.",
+                debug_message=response.content,
+                status_code=502
+            )
+    
```

Comment:
This change is to treat rate-limiting and invalid credentials as an error condition rather than an unusually formatted API response.